files/st2_task_base.py

```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/env python
+import json
+import os
+import re
+import subprocess
+import sys
+import traceback
+
+# import Bolt task helper
+sys.path.append(os.path.join(os.environ['PT__installdir'], 'python_task_helper', 'files'))
+from task_helper import TaskHelper, TaskError
+
+try:
+    # python 2
+    from urlparse import urlparse
+except ImportError:
+    # python 3
+    from urllib.parse import urlparse  # noqa
+
+# try to find a [ or { at the start of a line
+JSON_START_PATTERN = re.compile("(\\[|{)", re.MULTILINE)
+
+
+class St2TaskBase(TaskHelper):
+
+    def login(self, args):
+        self.api_key = args.get('api_key')
+        self.auth_token = args.get('auth_token')
+        self.username = args.get('username')
+        self.password = args.get('password')
+        # inherit environment variables from the Bolt context to preserve things
+        # like locale... otherwise we get errors from the StackStorm client.
+        self.env = os.environ
+
+        # prefer API key over auth tokens
+        if self.api_key:
+            self.env['ST2_API_KEY'] = self.api_key
+        elif self.auth_token:
+            self.env['ST2_AUTH_TOKEN'] = self.auth_token
+        elif self.username and self.password:
+            # auth on the command line with username/password
+            cmd = ['st2', 'auth', '--only-token', '-p', self.password, self.username]
+            stdout = subprocess.check_output(cmd)
+            self.env['ST2_AUTH_TOKEN'] = stdout.rstrip()
+        # else
+        #    assume auth token is written in client config for this user.
+        #    don't worry, if there is no auth we'll get an error
+
+    def scan_for_json(self, stdout):
+        # the output from st2 pack install doesn't print out in pure JSON, so
+        # look for the JSON in the output
```

Comment:
Is it still an issue?

Your issue https://github.com/StackStorm/st2/issues/4260 seems to be fixed here https://github.com/StackStorm/st2/pull/4394

Reply:
Removed the scanning portion. How it simply tries to parse as JSON starting at the first character. If that fails then the raw output string is returned.